lib/Parser/LineDataParser.php

```diff
@@ -101,4 +105,45 @@ public function parseListLine(string $line) : ?ListLine
 
         return null;
     }
+
+    /**
+     * @param string[] $lines
+     */
+    public function parseDefinitionList(array $lines) : DefinitionList
+    {
+        $definitionList     = new DefinitionList();
+        $definitionListTerm = null;
+
+        foreach ($lines as $key => $line) {
+            // term definition
+            if ($definitionListTerm !== null && substr($line, 0, 4) === '    ') {
+                $definition = trim($line);
+
+                $definitionListTerm->addDefinition($definition);
+
+            // non empty string
+            } elseif (trim($line) !== '') {
+                // we are starting a new term so if we have an existing
+                // term with definitions, add it to the definition list
+                if ($definitionListTerm !== null) {
+                    $definitionList->addTerm($definitionListTerm);
+                }
+
+                $parts = explode(':', trim($line));
+
+                $term = $parts[0];
+                unset($parts[0]);
+
+                $classifiers = array_map('trim', $parts);
+
+                $definitionListTerm = new DefinitionListTerm($term, $classifiers);
+
+            // last line
+            } elseif ($definitionListTerm !== null && trim($line) === '' && count($lines) - 1 === $key) {
+                $definitionList->addTerm($definitionListTerm);
```

Comment:
Would probably be better to have a `$definitionListTerms[]` in here, then make the `DefinitionList` from all terms assembled upfront

Reply:
I had it this way originally, or rather before I had the `DefinitionList` and `DefinitionListTerm` classes, I was just building up an array and passing the array around everywhere. Then I introduced the classes and it cleaned up the code a little to construct the data with an OO interface instead of using arrays. I'm find making those classes immutable though and building up the data to pass in through the constructor.